src/plugins/discover/public/utils/row_heights.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { IUiSettingsClient } from 'kibana/public';
+import { Storage } from '../../../kibana_utils/public';
+import { ROW_HEIGHT_OPTION } from '../../common';
+
+const ROW_HEIGHT_KEY = 'discover:dataGridRowHeight';
+const MIN_ROW_HEIGHT = -1;
+const MAX_ROW_HEIGHT = 20;
+
+interface DataGridOptionsRecord {
+  previousRowHeight: number;
+  previousConfigRowHeight: number;
+}
+
+const isValidRowHeight = (rowHeight?: number) => {
+  return (
+    typeof rowHeight === 'number' &&
+    !Number.isNaN(rowHeight) &&
```

Comment:
isn't `!Number.isNaN` sufficient to check?

Reply:
`rowHeight` might be undefined / null / ''. We have to check if it is a number type. Thx for your concern, it turns out we have to check here whether given rowHeight is integer.

![6456D792-DE67-4310-BDCF-81A5A22ECB00_4_5005_c](https://user-images.githubusercontent.com/39378793/149794915-e0e27e99-2c45-4320-a499-a5d2acbcfad3.jpeg)

Reply:
Improved `isInteger` check